src/cmd/troubleshoot/image_helper.go

```diff
@@ -0,0 +1,141 @@
+package troubleshoot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	removeSchemaRegex      = regexp.MustCompile("^.*//(.*)$")
+	removeApiEndpointRegex = regexp.MustCompile("^(.*)/[^/]*$")
+	registryRegex          = regexp.MustCompile(`^(.*)/linux.*$`)
+	imageRegex             = regexp.MustCompile(`^.*/(linux.*)$`)
+)
+
+func splitImageName(imageName string) (registry string, image string, version string, err error) {
```

Comment:
registry, image, version should be a struct (e.g. `imageInfo`) and should be in the return, not implicitly returned

Reply:
we already have `properties.go` and `image_paths.go` to check image fields. The logic here feels similar and we should combine it and move it to a location where it can be reused.

Reply:
This function has only be moved and is not part of this PR per-se. So I'd rather not mix general code cleanups with the purpose of this PR. If necessary, this kind of cleanups shall be a separate PR.

Reply:
I'd have no problems with smaller, general changes being in the same PR. Just large overhauls should be their own ones imo

Reply:
I agree here, if we already touch that code (does not matter if its changed or just moved) we can make it better in the same PR

Reply:
The whole `splitImage` code will be revisted in another PR because there seem to be other problems hidden. Will take this comment as input for that PR as well.